pkg/kn/commands/service/migrate_test.go

```diff
@@ -0,0 +1,161 @@
+//// Copyright © 2019 The Knative Authors
+////
+//// Licensed under the Apache License, Version 2.0 (the "License");
+//// you may not use this file except in compliance with the License.
+//// You may obtain a copy of the License at
+////
+////     http://www.apache.org/licenses/LICENSE-2.0
+////
+//// Unless required by applicable law or agreed to in writing, software
+//// distributed under the License is distributed on an "AS IS" BASIS,
+//// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//// See the License for the specific language governing permissions and
+//// limitations under the License.
+//
+package service
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	servinglib "knative.dev/client/pkg/serving"
+
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/wait"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	client_testing "k8s.io/client-go/testing"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+)
+
+func fakeMigrate(args []string, withExistingService bool, sync bool) (
+	action client_testing.Action,
+	service *v1alpha1.Service,
+	output string,
+	err error) {
+	knParams := &commands.KnParams{}
+	nrGetCalled := 0
+	cmd, fakeServing, buf := commands.CreateTestKnCommand(NewServiceCommand(knParams), knParams)
+	fakeServing.AddReactor("get", "services",
```

Comment:
Woudl it be possible to switch to the KnClient Mock testing framework as we use it in https://github.com/knative/client/blob/master/pkg/kn/commands/service/describe_test.go 

See also https://github.com/knative/client/issues/358 with the plan to migrate existing tests to the Mock Tests which would also ensure that Knative is accessed only via the `KnClient` abstraction. New code should already adapt to this "new" way of testing to save some work later on.

Reply:
Yes, change to use KnClient Mock testing framework